File: spring-integration/subscribable-channels/messaging-bridge/src/main/java/com/ddubson/integration/Application.java
package com.ddubson.integration;

import com.ddubson.CommandLineLogAdapter;
import com.ddubson.LogAdapter;
import com.ddubson.integration.gateways.PrinterGateway;
import com.ddubson.integration.services.PrintService;
import com.ddubson.integration.services.UppercasePrintService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.ApplicationArguments;
import org.springframework.boot.ApplicationRunner;
import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.ImportResource;
import org.springframework.messaging.Message;
import org.springframework.messaging.support.MessageBuilder;

import java.util.stream.Stream;

import static com.ddubson.ANSIColor.ANSI_WHITE;

@SpringBootApplication
//@ImportResource("integration-context.xml")
public class Application implements ApplicationRunner {
    private int numberOfMessages = 10;

    @Autowired
    private PrinterGateway gateway;

    @Override
    public void run(ApplicationArguments args) throws Exception {
        logAdapter().info("[1] Messaging Bridge", ANSI_WHITE);

        Stream.iterate(0, n -> n + 1).limit(numberOfMessages).forEach(i -> {
            Message<String> message = MessageBuilder.withPayload("Message " + i)
                    .build();
            this.gateway.print(message);
        });
    }

    public static void main(String[] args) {
        SpringApplication.run(Application.class, args);
    }

    @Bean
    public LogAdapter logAdapter() {
        return new CommandLineLogAdapter();
    }

    @Bean
    public PrintService printService() {
        return new PrintService(logAdapter());
    }

    @Bean
    public UppercasePrintService uppercasePrintService() {
        return new UppercasePrintService(logAdapter());
    }
}

File: spring-integration/subscribable-channels/messaging-bridge/src/main/kotlin/com/ddubson/integration/Application.kt
package com.ddubson.integration

import com.ddubson.ANSIColor
import com.ddubson.CommandLineLogAdapter
import com.ddubson.LogAdapter
import org.springframework.boot.ApplicationArguments
import org.springframework.boot.ApplicationRunner
import org.springframework.boot.SpringApplication
import org.springframework.boot.autoconfigure.SpringBootApplication
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.ImportResource
import org.springframework.messaging.support.MessageBuilder

@SpringBootApplication
@ImportResource("integration-context.xml")
class Application(private val gateway: PrinterGateway) : ApplicationRunner{
    private val numberOfMessages: Int = 10

    override fun run(args: ApplicationArguments?) {
        logAdapter().info("[1] Messaging Bridge", ANSIColor.ANSI_WHITE)

        (0..numberOfMessages).forEach {
            val message = MessageBuilder.withPayload("Message $it").build()
            gateway.print(message)
        }
    }

    @Bean
    fun logAdapter(): LogAdapter = CommandLineLogAdapter()
}

fun main(args: Array<String>) {
    SpringApplication.run(Application::class.java, *args)
}